File: demos/jans-chip/android/app/src/main/java/io/jans/chip/dao/OidcClientDao.java
package io.jans.chip.dao;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

import io.jans.chip.modal.OIDCClient;
@Dao
public interface OidcClientDao {
    @Insert
    void insert(OIDCClient oidcClient);

    @Update
    void update(OIDCClient oidcClient);

    @Query("SELECT * FROM OIDC_CLIENT")
    List<OIDCClient> getAll();
    @Query("DELETE FROM OIDC_CLIENT")
    void deleteAll();
}

File: demos/jans-chip/android/app/src/main/java/io/jans/chip/dao/OidcClientDao.kt
package io.jans.chip.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update
import io.jans.chip.model.OIDCClient
@Dao
interface OidcClientDao {
    @Insert
    fun insert(oidcClient: OIDCClient)

    @Update
    fun update(oidcClient: OIDCClient)

    @Query("SELECT * FROM OIDC_CLIENT")
    fun getAll(): List<OIDCClient>

    @Query("DELETE FROM OIDC_CLIENT")
    fun deleteAll()
}